client/app/dashboard/setup/books/AddBookModal.tsx

```diff
@@ -0,0 +1,141 @@
+import { Modal } from "@/components";
+import { SetupBook } from "@/hooks";
+import { useGetBook } from "@/hooks/useGetBook";
+import { Spinner } from "@jecfe/react-design-system";
+import Image from "next/image";
+import { Dispatch, ReactNode, SetStateAction, useEffect, useMemo } from "react";
+
+type Props = {
+  isbn?: string;
+  addBook: (book: SetupBook) => void;
+  showModal: boolean;
+  setShowModal: Dispatch<SetStateAction<boolean>>;
+  setPassingIsbn: Dispatch<SetStateAction<string | undefined>>;
+  setCurrentIsbn: Dispatch<SetStateAction<string | undefined>>;
+};
+
+export function AddBookModal({
+  isbn,
+  addBook,
+  showModal,
+  setShowModal,
+  setPassingIsbn,
+  setCurrentIsbn,
+}: Props) {
+  const { data, isLoading } = useGetBook(isbn);
+  const setupBook: SetupBook | undefined = useMemo(():
+    | SetupBook
+    | undefined => {
+    if (data === undefined) {
+      return undefined;
+    }
+    return {
+      isbn: data.isbn as string,
+      name: data.name as string,
+      release: data.release as string,
+      picture: data.picture as string,
+      pageCount: data.pageCount,
+      authors: data.authors as string[] | undefined,
+      subjects: data.subjects as string[] | undefined,
+    };
+  }, [data]);
+
+  useEffect(() => console.log(setupBook), [setupBook]);
+  return (
+    <Modal
+      isOpen={showModal}
+      onClose={() => {
+        setPassingIsbn(undefined);
+        setShowModal(false);
+      }}
+      onConfirm={() => {
+        if (setupBook === undefined) {
+          return;
+        }
+        addBook(setupBook);
+        setPassingIsbn(undefined);
+        setShowModal(false);
+        setCurrentIsbn(undefined);
+      }}
+      error=""
```

Comment:
Will need to actually handle error conditions -> 404 is the most likely, anything else should be Unknown 

Perhaps Toasts, who doesnt love a good toast